src/main/scala/devices/spi/SPIPhysical.scala

```diff
@@ -99,37 +101,25 @@ class SPIPhysical(c: SPIParamsBase) extends Module {
 
   val rxd = Cat(io.port.dq.reverse.map(_.i))
   val rxd_delayed = Vec(Seq.fill(io.port.dq.size)(false.B))
-  //0.U(io.port.dq.size.W).toBools
+
   //Adding fine-granularity delay buffers on the received data
+  val fpga_syn = false   //Change this to true for FPGA simulation
+  if (fpga_syn != true){
+    val fine_grain_delay = Seq.fill(4){Module(new BlackBoxDelayBuffer_tsmc28_INVD4BWP12T35P140_MUX2D2BWP12T35P140())}
+    for (j <- 0 to 3 ){ 
+      fine_grain_delay(j).io.in := rxd(j)
```

Comment:
this is fine, but an even more Scalay- way of doing it would be:

((fine_grain_delay zip rxd) zip rxd_delayed).foreach { case ((fgd, r), rd) => 
  fgd.io.in := r
  fgd.io.sel := io.ctrl.extradel.fine
  rd := fgd.io.mux_out
}
```

(someone gave me this sort of comment on the first (and close to last) `for` loop I ever wrote in scala...) This has the nice property that it works no matter how many `fine_grain_delay` you have. (Though it can also cause problems if one list is unexpectedly shorter than the other, so pick your poison).